src/llama-graph.cpp

```diff
@@ -278,7 +278,61 @@ void llm_graph_input_cross_embd::set_input(const llama_ubatch * ubatch) {
 
 void llm_graph_input_attn_no_cache::set_input(const llama_ubatch * ubatch) {
     if (kq_mask) {
-        if (cparams.causal_attn) {
+        // Check if we're using sliding window attention
+        if (n_swa > 0) {
+            const int64_t n_tokens     = ubatch->n_tokens;
+            const int64_t n_seq_tokens = ubatch->n_seq_tokens;
```

Comment:
This branch is actually `non-causal` attention + sliding window. So merge it with the existing implementation below.